validator/executable/validator.go

```diff
@@ -0,0 +1,34 @@
+package executablevalidator
+
+import (
+	"os/exec"
+)
+
+func NewExecutableValidator(path string) (*executableValidator, error) {
+	return &executableValidator{executable: path}, nil
+}
+
+type executableValidator struct {
```

Comment:
This type should be exported for documentation and because it's hard to deal with unexported types (say you wanted to embed it as a field in your struct.